attachment_synchronize/views/attachment_queue_views.xml

```diff
@@ -10,18 +10,19 @@
                 <field
                     name="task_id"
                     domain="[('method_type', '!=', 'import')]"
-                    attrs="{'required': [('file_type', '=', 'export')], 'readonly': [('method_type', '=', 'import')]}"
+                    required="file_type == 'export'"
+                    readonly="method_type == 'import'"
                 />
                 <field name="fs_storage_id" />
             </xpath>
         </field>
     </record>
 
-    <record id="view_attachment_queue_tree" model="ir.ui.view">
+    <record id="view_attachment_queue_list" model="ir.ui.view">
```

Comment:
The recommendation is not to rename view ids in this case. The reason for this is that Odoo does not handle renamed view renames very well during a migration. The migration will halt if the old view is not valid, which is likely the case here because it tries to fetch a `tree` node. Renaming is ok if you provide a pre-migration script renaming the ir.model.data record.

Reply:
Hum, I did not know this one, thanks...
Here, I already rename the base view in the dependency `attachment_queue`, where we must have this migration issue.
I think I will go with the migration script then, in both module.

Reply:
@StefanRijnhart I did the following test (with attachment_queue, that has been merged already but has the same change on list view xml id).
On a demo database with the module installed, I renamed the `ir.model.data` from view_attachment_queue_list to view_attachment_queue_tree
I update the arch_db (by sql query) to the old one with the `<tree>` tags.
Update the latest_version number to 17.0.1.0.0.
Update my the module attachment_queue, to simulate a migration.
=> The old view is deleted, the new one created and I do not have any crash.

Are you sure the migration script is necessary ? Of course, I may have missed something as I did not run a migration, just tried to simulate it.
If you are quite sure, I'll add  the scripts, in both modules, what do you think ?

Reply:
@StefanRijnhart Thanks for your opinion and for pointing the guidline.
It was a real question on my side, I am discovering this issue, so I trust you know better than me indeed.
I'll add the migration scripts (seems better than revert the xml_id because of the merged dependency `attachment_queue`)